### DISCUSSION TITLE
ESP messages between two Devices not exchanged although the IPSEC tunnel seems to be opened (Version 2)

### DISCUSSION THREAD

**Discussion by carros06** [strongswan/strongswan#976] [Q&A] [answered]
Hello
My previous discussion did not provide consistent information. I am very sorry about that.
That is why I create a new one with consistent information. I hope the information is now consistent

The Device strongswan IPSEC.conf configuration below is used for several purposes:
Rule 172.20.0.60-tunnel for communication between Windows and a Linux Device supporting Strongswan
Rule 172.20.0.60-tunnel-P2P for communication between two Linux devices supporting Strongswan.

If Windows closes the IPSEC tunnel, the IPSEC tunnel is not automatically restarted by the device when a message is sent from the device to Windows. 
To solve this problem I change "auto=start" by "auto=route" in 192.168.8.2-tunnel rule.
But after this change the device to device to device ESP communication is not exchanged although the IPSEC tunnel seems to be opened as shown by IPSEC status below.

Do you have some idea of the reason of this problem ?

The Device strongswan IPSEC.conf configuration is:

config setup
    charondebug="ike 1"

conn %default
    ikelifetime=2879m
    lifetime=24h
    authby=secret
    ike=aes128-sha256-modp2048!
    esp=aes128-sha256! # Default encryption mode
    # for a larger scope and windows compatibility keep only ikev1
    # When both connection ikev1 and ikev2 started (tested in transport mode), it looks like windows do not work in ikev2, either
    # with kooka initiator or windows initiator => removed ikev2
    # windows doc: IKEv2 for PSK not supported
    keyexchange=ikev1

conn 172.20.0.60-passthroughHTTPS
    type=passthrough
    left=127.0.0.1
    leftsubnet=172.20.0.51/32[tcp/443]
    rightsubnet=172.20.0.60/32
    auto=route

conn 172.20.0.60-transport
    type=transport
    left=172.20.0.51
    right=172.20.0.60
    auto=start

conn 172.20.0.60-tunnel # PC to Device Network
    type=tunnel
    left=172.20.0.51
    leftsubnet=169.20.0.51/16
    right=172.20.0.60
    rightsubnet=172.20.0.60/32
    auto=start

conn 172.20.0.60-tunnel-P2P # Device to Device
    keyexchange=ikev2
    type=tunnel
    left=172.20.0.51
    leftsubnet=172.20.0.51/32
    right=172.20.0.60
    rightsubnet=172.20.0.60/32
    auto=start

#Priorities of drop policies are lower than that of actually negociated policies
conn 172.20.0.60-tunnel-P2P-drop # Device to Device drop policy
    type=drop
    left=172.20.0.51
    leftsubnet=172.20.0.51/32
    right=172.20.0.60
    rightsubnet=172.20.0.60/32[tcp/502]
    auto=route
	
IPSEC status on Device:

Shunted Connections:
172.20.0.60-passthroughHTTPS:  172.20.0.51/32[tcp/https] === 172.20.0.60/32 PASS
172.20.0.60-tunnel-P2P-drop:  172.20.0.51/32 === 172.20.0.60/32[tcp/502] DROP
Routed Connections:
172.20.0.60-tunnel{1}:  ROUTED, TUNNEL, reqid 1
172.20.0.60-tunnel{1}:   169.20.0.0/16 === 172.20.0.60/32
Security Associations (2 up, 0 connecting):
172.20.0.60-transport[4]: ESTABLISHED 2 minutes ago, 172.20.0.51[172.20.0.51]...172.20.0.60[172.20.0.60]
172.20.0.60-transport{6}:  INSTALLED, TRANSPORT, reqid 2, ESP SPIs: c1dd848c_i c67b91a0_o
172.20.0.60-transport{6}:   172.20.0.51/32 === 172.20.0.60/32
172.20.0.60-tunnel-P2P[3]: ESTABLISHED 2 minutes ago, 172.20.0.51[172.20.0.51]...172.20.0.60[172.20.0.60]
172.20.0.60-tunnel-P2P{5}:  INSTALLED, TUNNEL, reqid 2, ESP SPIs: c522c484_i c1668708_o
172.20.0.60-tunnel-P2P{5}:   172.20.0.51/32 === 172.20.0.60/32

Many Thanks

**Comment by tobiasbrunner** [comment]
There is still stuff that doesn't make sense. Why is the IP of the Windows and the Linux host the same (`right` is the same in all connections)? What's the point of `172.20.0.60-transport` that uses IKEv1 and transport mode and the duplicate `conn 172.20.0.60-tunnel-P2P` that uses IKEv2 but tunnel mode? You mentioned something about using `auto=route` but other than the shunts there is no config that actually uses that (while the status output does show that it was set for `172.20.0.60-tunnel`). So you maybe should start from scratch and really think about what it is exactly you are trying to do. Then you should read #196 and then maybe come back with more detailed and precise information what the actual problem is.

**Reply by carros06** [reply]
OK, I am going to provide you more precise information and try to be more clear on the objectives

**Comment by carros06** [comment]
[ipsecconf.txt](https://github.com/strongswan/strongswan/files/8388396/ipsecconf.txt)

[ipsecstatus.txt](https://github.com/strongswan/strongswan/files/8388338/ipsecstatus.txt)

[architecture2.pptx](https://github.com/strongswan/strongswan/files/8388322/architecture2.pptx)


I attach somme files  to help to the understanding:
*architecture2.ppt
*ipsecconf.txt
*ipsecstatus.txt


All those files are taken on MODULE 2 (see below).
Note that MODULE 1 has similar parameters

The strongswan modules that I call MODULE1 (@IP = 172.20.0.60) and MODULE2 (@IP = 172.20.0.51) are doing IPSEC and IP FORWARDING through IPTABLES:


172.20.0.60-tunnel-P2P rule is used for Peer to Peer communications initiated by a DEVICE2 that is behind MODULE2 to a DEVICE1 that is behind MODULE1 
The communication is done using ikeV2 and NAT address translation. 
NAT translation in MODULE2 for the message received from DEVICE2:
iptables -t nat -A POSTROUTING -o eth0 -p tcp -s 169.20.0.45 --dport xxx -j SNAT --to-source 172.20.0.51
NAT translation in MODULE2 for the message received from MODULE1 and sent to DEVICE2:
 iptables -t nat -A PREROUTING -i eth0 -p tcp -d 172.20.0.51 --dport xxx -j DNAT --to-destination 169.20.0.45
 iptables -t nat -A POSTROUTING -o eth1 -p tcp -d 169.20.0.45 --dport xxx -j SNAT --to-source 169.20.0.51

The other rules are here for the other use cases:
xxxx-transport rule is here for communications initiated by a Windows PC (IKEv1) to the local services of MODULE2
xxxxx-tunnel rule is here for communications initiated by a Windows PC (IKEv1) to DEVICE2 that is behind MODULE2 on 169.20.x.x subnet


Now I need to have a strongswann rule for communications iniated by DEVICE2 that is behind MODULE2 on 169.20.xx subnet to Windows PC (IKEv1) and 
this communication should open automatically the tunnel.

For this purpose I change the auto=start to auto=route in the rule:

conn ${ipAddress}-tunnel # PC to Device Network
    type=tunnel
    left=${FRONT_IPV4}
    leftsubnet=${BACK_SUBNET}
    right=${ipAddress}
    rightsubnet=${ipAddress}/32
    auto=start
    ${espMode:-}

But with this change the peer to peer communication is no more working. The ESP communication is not exchanged although the IPSEC tunnel seems to be opened as shown in ipsecstatus.txt.

Do you have an idea of the reason ?

**Reply by tobiasbrunner** [reply]
> xxxx-transport rule is here for communications initiated by a Windows PC (IKEv1) to the local services of MODULE2

Why is `right` set to MODULE1's IP then? Shouldn't it be `172.20.0.105` or `%any`? What you configured/generated is a host-to-host connection between MODULE1 and MODULE2.

> xxxxx-tunnel rule is here for communications initiated by a Windows PC (IKEv1) to DEVICE2 that is behind MODULE2 on 169.20.x.x subnet

Same as above but additionally here `left|rightsubnet` make no sense either (`rightsubnet` should probably not be specified and `leftsubnet` should probably be the IP of DEVICE2, unless you want to involve some NAT again).

**Reply by carros06** [reply]
In this example, there is no configuration of an IPSEC link between Windows and MODULE2. Only the IPSEC link between MODULE 2 and MODULE 1 is configured.
But MODULE 2 does not know if the remote is a Windows PC or a MODULE1 device.
So the same IPSEC rules are always generated by the each MODULE whatever is the remote.
In this example the remote is MODULE1. So all the rules are generated by MODULE2 but only 172.20.0.60-tunnel-P2P rule should be used

**Reply by tobiasbrunner** [reply]
Sorry, but that makes no sense whatsoever.

**Reply by carros06** [reply]
In this example, there is no configuration of an IPSEC link between Windows and MODULE2. Only the IPSEC link between MODULE 2 and MODULE 1 is configured.
But MODULE 2 does not know if the remote is a Windows PC or a MODULE1 device.
So the same IPSEC rules are always generated by the each MODULE whatever is the remote.
In this example the remote is MODULE1. So all the rules are generated by MODULE2 but only 172.20.0.60-tunnel-P2P rule should be used

**Reply by carros06** [reply]
I probably have some troubles to explain

**Reply by carros06** [reply]
Sorry I am not clear on the reason why it makes no sense:

In this configuration only the xxxx-tunnel-P2P is necessary to do DEVICE2 (through MODULE2) to DEVICE1 (through MODULE1) communications.
As I try to explain the same IPSEC rules are always generated by the each MODULE whatever is the remote.
The reason of each rule is the following:
xxxx-transport rule is here for communications initiated by a Windows PC (IKEv1) to the local services of MODULE2
xxxxx-tunnel rule is here for communications initiated by a Windows PC (IKEv1) to a device that is behind MODULE2 on 169.20.x.x subnet (for example DEVICE2)
xxxx-tunnel-P2P is here for communications between MODULE2 and MODULE1 (IKEV2)

What is your recommendation ?
Do you need additional information ?

**Reply by tobiasbrunner** [reply]
Configure only the connections for the traffic you actually want to protect, fix the IPs and traffic selectors (see my previous comment) etc.

**Comment by carros06** [comment, marked as answer]
Unfortunately there is no way on our product for the user to configure the type of remote (Windows, Device).
So we have no way to configure only the connections for the traffic we actually want to protect.
The problem we detected with our current IPSEC rules settings is that there is no way to open automatically the iPSEC tunnel if it is closed for example by remote Windows.
For this problem, we tried to change auto=start to auto=route in conn xxxxx-tunnel. But with this change the xxxx-tunnel-P2P communication is no more working. 
Do you have recommendation to be able to open automatically a closed IPSEC tunnel with the IPSEC rules that I described to you by another mean than setting “auto=route”?

**Reply by carros06** [reply]
Hello, Do you have some response to provide to my last message ?
